sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -102,7 +99,7 @@ def __init__(self, client_id, client_credential=None, **kwargs):
             else:
                 self._cache = msal.TokenCache()
 
-        self._adapter = kwargs.pop("msal_adapter", None) or MsalTransportAdapter(**kwargs)
+        self._client = kwargs.pop("msal_adapter", None) or MsalClient(**kwargs)
```

Comment:
Why we still use msal_adapter?

Reply:
Compatibility. It's not documented and nobody should be using it but if they are, I don't want to break them.

Reply:
It seems to me MsalClient is not fully compatible with MsalTransportAdapter?

Reply:
That's a good point. It could work but we can't guarantee it. It would be nice to remove the argument. I intended it for tests but none use it.

What do you think @johanste, is it important to maintain this argument's current behavior? This is an internal base class. The argument isn't documented and is referenced only on this line. I'd be shocked to learn someone figured out how to use it and found value in doing so. However, the public API passes `kwargs` down to this class, so in principle someone could depend on the current behavior.

Reply:
Was `MsalTransportAdapter` a public class?

Reply:
No.

Reply:
So the only way(s) someone will hit this is if they use an internal, undocumented class (or something that behaves like an internal, undocumented class) and pass it as an undocumented keyword argument. I would rename the undocumented argument rather than keeping it the same for "compatiblity's sake" (especially since it isn't actually compatible).